### ISSUE TITLE
Feature: 使用bot发送消息来切换模型

### ISSUE THREAD

**Issue by yi03** [koishijs/novelai-bot#140]
### Scope

Stable Diffusion WebUI (AUTOMATIC1111)

### Describe the problem related to the feature request

本地有很多不同的模型，只能bot管理者手动切换，有些时候很不方便。

### Describe the solution you'd like

在群里向bot发送切换模型的指令后，bot切换模型。

**Comment by MaikoTan** [comment]
Since many sd-webui provider are individuals that they don't have higher spec computer. Switching models frequently causes downtime of the webui extremely long.
So I think if there were a command doing the such thing, it would be better to require an authority of level 2 or 3.
As always you could change that with `commands` plugin. Anyone thoughts?

自从很多 sd-webui 提供者是个人，没有高配置电脑。频繁地切换模型会造成 webui 极其长的停机时间。
所以我想如果有一个指令做了这件事情，要求一个 2 或 3 的权威会更好。和往常一样你可以通过`指令`插件改变。有人想法？

**Comment by yi03** [comment]
> Since many sd-webui provider are individuals that they don't have higher spec computer. Switching models frequently causes downtime of the webui extremely long. So I think if there were a command doing the such thing, it would be better to require an authority of level 2 or 3. As always you could change that with `commands` plugin. Anyone thoughts?
> 
> 自从很多 sd-webui 提供者是个人，没有高配置电脑。频繁地切换模型会造成 webui 极其长的停机时间。 所以我想如果有一个指令做了这件事情，要求一个 2 或 3 的权威会更好。和往常一样你可以通过`指令`插件改变。有人想法？

其实现在sd-switch插件已经基本满足了我的需求了。我切换2GB大小的`DreamBooth`模型大概需要五六秒，由于现在我的群玩ai的群友已经很少了，多个群友同时玩的次数不多，所以我没设置什么权限限制。当然我这种情况应该比较少，还是应该有一个权限限制。通过`commads`插件来改变我觉得挺好的。
另外除了`DreamBooth`模型之外还有`Textual Inversion`, `Hypernetwork`, `Lora`这些小模型，这几个小模型在最新版的sd-webui里都可以在prompt里来选择，比如在prompt里输入`<lora:amberGenshinImpact_flexibleV1:1>`就表示选择了一个名叫`amberGenshinImpact_flexibleV1`的`Lora`模型，权重是1。所以需要一个列出当前所有可用`Textual Inversion`, `Hypernetwork`, `Lora`模型的功能。这三种模型里我个人的话`Lora`模型用的最多，有时会用`Textual Inversion`模型，几乎不用`Hypernetwork`模型。

**Comment by MaikoTan** [comment]
> 其实现在sd-switch插件已经基本满足了我的需求了。

In fact, our implementations would be different with `sd-switch`. I would add documents for it soon after we succesfully distrbute this feature.

> 通过`commads`插件来改变我觉得挺好的。

OK, I would walk this way.

> 另外除了`DreamBooth`模型之外还有`Textual Inversion`, `Hypernetwork`, `Lora`这些小模型，这几个小模型在最新版的sd-webui里都可以在prompt里来选择，比如在prompt里输入`<lora:amberGenshinImpact_flexibleV1:1>`就表示选择了一个名叫`amberGenshinImpact_flexibleV1`的`Lora`模型，权重是1。

Currently the `<` and `>` wouldn't be convert into `&lt` and `&gt`? I thought it would be the issue that should be fixed before!

> 所以需要一个列出当前所有可用`Textual Inversion`, `Hypernetwork`, `Lora`模型的功能。

That is another feature request I think.

**Comment by yi03** [comment]
> 

Yes. Currently the `<` and `>` would be convert into `&lt` and `&gt`. #161
> That is another feature request I think.

OK. I will open an another issue.

**Comment by MaikoTan** [comment]
> Yes. Currently the < and > would be convert into &lt and &gt. https://github.com/koishijs/novelai-bot/issues/161

Oh, okay.
Too many works in these days that I even forgot this issue that I commented.

**Comment by s567901** [comment]
附议，希望能通过向bot发送切换模型😀

**Comment by Ninzore** [comment]
在做了

**Comment by MirrorCY** [comment]
感觉现在这个项目已经很重了，这个需求我建议放到 sd-switch 实现，你觉得如何 @MaikoTan 

顺便，有任何与这个功能相关的需求可以到这里提出 https://github.com/MirrorCY/sd-switch/issues
